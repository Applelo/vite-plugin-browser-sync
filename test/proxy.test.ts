import { resolve } from 'path'
import { Browser, chromium, Page } from 'playwright'
import {
  it,
  expect,
  describe,
  afterAll,
  beforeAll,
  afterEach,
  beforeEach
} from 'vitest'
import { createServer } from 'vite'
import VitePluginBrowserSync from '../src'

let browser: Browser
let page: Page

beforeAll(async () => {
  browser = await chromium.launch()
})
afterAll(async () => {
  await browser.close()
})
beforeEach(async () => {
  page = await browser.newPage()
})
afterEach(async () => {
  await page.close()
})

describe('proxy option', () => {
  it('default', async () => {
    const server = await createServer({
      // any valid user config options, plus `mode` and `configFile`
      configFile: false,
      root: resolve(__dirname, './../demo'),
      plugins: [VitePluginBrowserSync()]
    })
    await server.listen()

    // need to use playwright to test the proxy
    await page.goto('http://localhost:3000')
    const content = await page.content()

    server.close()
    expect(content).matchSnapshot()
  })
})
