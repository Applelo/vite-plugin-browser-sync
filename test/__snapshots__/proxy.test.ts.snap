// Vitest Snapshot v1

exports[`proxy option > default 1`] = `
"<!DOCTYPE html><html lang=\\"en\\"><head>
    <script type=\\"module\\" src=\\"/@vite/client\\"></script>

    <meta charset=\\"UTF-8\\">
    <link rel=\\"icon\\" type=\\"image/svg+xml\\" href=\\"/vite.svg\\">
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\">
    <title>Vite Plugin Browser Sync</title>
  <style type=\\"text/css\\">:root {
  font-family: Inter, Avenir, Helvetica, Arial, sans-serif;
  font-size: 16px;
  line-height: 24px;
  font-weight: 400;

  color-scheme: light dark;
  color: #dedede;
  background-color: #121212;

  font-synthesis: none;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  -webkit-text-size-adjust: 100%;
}

body {
  margin: 0;
  padding: 5vw;
  min-height: 200vh;
}

h1 {
  font-size: 3.2em;
  line-height: 1.1;
}
</style></head>
  <body><script id=\\"__bs_script__\\">//<![CDATA[
    document.write(\\"<script async src='/browser-sync/browser-sync-client.js?v=2.27.10'><\\\\/script>\\".replace(\\"HOST\\", location.hostname));
//]]></script><script async=\\"\\" src=\\"/browser-sync/browser-sync-client.js?v=2.27.10\\"></script>

    <h1>Vite Plugin Browser Sync</h1>

    <form id=\\"form\\">
      <div>
        <label for=\\"firstname\\">First Name</label>
        <input type=\\"text\\" id=\\"firstname\\" name=\\"firstname\\">
      </div>
      <div>
        <label for=\\"lastname\\">Last Name</label>
        <input type=\\"text\\" id=\\"lastname\\" name=\\"lastname\\">
      </div>
      <input type=\\"submit\\" value=\\"Submit\\">
    </form>
    <p id=\\"message\\"></p>

    <script type=\\"module\\" src=\\"/src/main.ts\\"></script>
  

<div id=\\"__bs_notify__\\" style=\\"display: block; padding: 15px; font-family: sans-serif; position: fixed; font-size: 0.9em; z-index: 9999; right: 0px; top: 0px; border-bottom-left-radius: 5px; background-color: rgb(27, 32, 50); margin: 0px; color: white; text-align: center; pointer-events: none;\\">vite-plugin-browser-sync: connected</div></body></html>"
`;
